src/readers/waffle/waffle.js

```diff
@@ -154,7 +102,80 @@ var WSReader = Reader.extend({
    */
   getData: function () {
     return this._data;
+  },
+
+  _readCallbackSuccess: function (p, path, query, resp) {
+
+    if (!resp) {
+      utils.error("Empty json: " + path);
+      p.reject({
+        'message' : this.ERROR_RESPONSE,
+        'data': path
+      });
+      return;
+    }
+
+    //format data
+    resp = utils.mapRows(this._uzip(resp.data || resp), this._parsers);
+
+    //cache and resolve
+    FILE_CACHED[path] = resp;
+
+    this._parse(p, query, resp);
+    FILE_REQUESTED[path] = void 0;
+  },
+
+  _readCallbackError: function (p, path, query, resp) {
+    p.reject({
+      'message' : this.ERROR_NETWORK,
+      'data': path
+    });
+  },
+
+  _uzip: function (table) {
+    var rows = table.rows;
+    var headers = table.headers;
+    var result = new Array(rows.length);
+    // unwrap compact data into json collection
+    for (var i = 0; i < rows.length; i++) {
+      result[i] = {};
+      for (var j = 0; j < headers.length; j++) {
+        if (typeof rows[i][j] == 'undefined' || rows[i][j] === null) {
+          result[i][headers[j]] = '';
```

Comment:
Wouldn't it be more readable to name `j` as `headerIndex`
and extract `headers[headerIndex]` into variable named `header`

`var header = headers[headerIndex];`

same with the rows